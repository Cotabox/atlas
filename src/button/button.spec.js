import { shallowMount } from '@vue/test-utils';
import Button from './button.vue';

describe('Button', () => {
  let wrapper;
  describe('when is a default button', () => {
    beforeEach(() => {
      wrapper = shallowMount(Button);
    });

    it('should be match with the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('when is a submit button', () => {
    beforeEach(() => {
      wrapper = shallowMount(Button, {
        propsData: {
          type: 'submit',
        },
      });
    });

    it('should match with the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('when pass an onClick Function', () => {
    const onClick = jest.fn();

    beforeEach(() => {
      wrapper = shallowMount(Button, {
        propsData: {
          onClick,
        },
      });
    });

    describe('when button is clicked', () => {
      it('should call the onclick function', () => {
        wrapper.findAll('button').trigger('click');
        expect(onClick).toBeCalled();
      });
    });
  });
});
