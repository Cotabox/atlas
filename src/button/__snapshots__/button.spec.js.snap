// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when is a default button should be match with the snapshot 1`] = `<button class="a-button a-button--default"></button>`;

exports[`Button when is a submit button should match with the snapshot 1`] = `<button class="a-button a-button--submit"></button>`;
